mautic/mautic
# Add a way to directly modify the html code in Advanced mode

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [no]
| New feature/enhancement? (use the a.x branch)      | [yes]
| Deprecations?                          | [no]
| BC breaks? (use the c.x branch)        | [no]
| Automated tests included?              | [no] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes [#10769](https://github.com/mautic/mautic/issues/10769) 



#### Description:
If GrapesJS is enabled, a simple textarea (emailform_customHtml)  field is displayed in the advanced tab that can be used to to directly add HTML. The textarea is only visible if code mode was selected. The builder button to launch GrapesJS is disabled.

![image](https://user-images.githubusercontent.com/96085911/158850048-e01d4be5-013d-499d-aad9-8f5d1f7aa839.png)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new email
3. Select the code mode email type
4. The builder button is disabled
5. Select the advanced tab
6. The HTML Code field is displayed




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10976"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
